extract_test.go

```diff
@@ -106,25 +108,15 @@ var ExtractCases = []struct {
 	}},
 }
 
-func TestExtract(t *testing.T) {
+func testFunc(t *testing.T, innerFunc func(*testing.T, io.Reader, string, testCase)) {
```

Comment:
This kind of refactoring is totally uncalled for and makes the code more difficult to read.

Rule of Thumb. Keep the Test functions simple. I'm making another Pull request on your repo to show you.